packages/api/src/command/webhook/fitbit.ts

```diff
@@ -30,64 +30,82 @@ interface Entry {
 const log = Util.log(`Fitbit Webhook`);
 
 export const processData = async (data: FitbitWebhook): Promise<void> => {
-  console.log("Starting to process the webhook");
+  console.log("Starting to process a Fitbit webhook.");
 
   const connectedUsersAndData = await Promise.all(
     data.map(async d => {
       const { collectionType, date, ownerId: fitbitUserId } = d;
-      const connectedUser = await getConnectedUserByTokenOrFail(
-        ProviderSource.fitbit,
-        fitbitUserId
-      );
+      try {
+        const connectedUsers = await getConnectedUsersByTokenOrFail(
+          ProviderSource.fitbit,
+          fitbitUserId
+        );
+
+        const userFitbitData = await mapData(collectionType, connectedUsers[0], date);
+        const returnData: Entry[] = [];
 
-      const userFitbitData = await mapData(collectionType, connectedUser, date);
-      return { cxId: connectedUser.cxId, userId: connectedUser.id, typedData: userFitbitData };
+        connectedUsers.forEach(connectedUser => {
+          returnData.push({
+            cxId: connectedUser.cxId,
+            userId: connectedUser.id,
+            typedData: userFitbitData,
+          });
+        });
+        return returnData;
+      } catch (err) {
+        console.log(`Failed to collect and map Fitbit data on a webhook. Cause: ${err}`);
+        capture.error(err, { extra: { context: `fitbit.processData`, err } });
+      }
```

Comment:
Why are continuing to process the data if a mapping fails here now?

I guess the idea is it's better than losing the data?

Reply:
Yea, I think in this case it's better to continue execution and send whatever data we're able to process